erts/emulator/nifs/win32/win_prim_file.c

```diff
@@ -716,39 +716,90 @@ static int is_executable_file(const efile_path_t *path) {
     return 0;
 }
 
-/* Returns whether the path refers to a link-like object, e.g. a junction
+/* Returns whether the handle refers to a link-like object, e.g. a junction
  * point, symbolic link, or mounted folder. */
-static int is_name_surrogate(const efile_path_t *path) {
-    HANDLE handle;
-    int result;
+static int is_handle_surrogate(HANDLE handle) {
+    REPARSE_GUID_DATA_BUFFER reparse_buffer;
+    DWORD unused_length;
+    BOOL success;
 
-    handle = CreateFileW((const WCHAR*)path->data, GENERIC_READ,
-                         FILE_SHARE_FLAGS, NULL, OPEN_EXISTING,
-                         FILE_FLAG_OPEN_REPARSE_POINT |
-                         FILE_FLAG_BACKUP_SEMANTICS,
-                         NULL);
-    result = 0;
+    success = DeviceIoControl(handle,
+            FSCTL_GET_REPARSE_POINT, NULL, 0,
+            &reparse_buffer, sizeof(reparse_buffer),
+            &unused_length, NULL);
 
-    if(handle != INVALID_HANDLE_VALUE) {
-        REPARSE_GUID_DATA_BUFFER reparse_buffer;
-        DWORD unused_length;
-        BOOL success;
-
-        success = DeviceIoControl(handle,
-                                  FSCTL_GET_REPARSE_POINT, NULL, 0,
-                                  &reparse_buffer, sizeof(reparse_buffer),
-                                  &unused_length, NULL);
-
-        /* ERROR_MORE_DATA is tolerated since we're guaranteed to have filled
-         * the field we want. */
-        if(success || GetLastError() == ERROR_MORE_DATA) {
-            result = IsReparseTagNameSurrogate(reparse_buffer.ReparseTag);
+    /* ERROR_MORE_DATA is tolerated since we're guaranteed to have filled
+     * the field we want. */
+    if(success || GetLastError() == ERROR_MORE_DATA) {
+        return IsReparseTagNameSurrogate(reparse_buffer.ReparseTag);
+    }
+
+    return 0;
+}
+
+static void build_file_info(BY_HANDLE_FILE_INFORMATION *native_file_info, efile_path_t *path, int is_link, efile_fileinfo_t *result) {
```

Comment:
`path` isn't modified here, so it should be `const`.
```suggestion
static void build_file_info(BY_HANDLE_FILE_INFORMATION *native_file_info, const efile_path_t *path, int is_link, efile_fileinfo_t *result) {
```